subsys/net/l2/wifi/wifi_shell.c

```diff
@@ -531,6 +531,40 @@ static int cmd_wifi_ps_mode(const struct shell *sh, size_t argc, char *argv[])
 	return 0;
 }
 
+static int cmd_wifi_ps_timeout(const struct shell *sh, size_t argc, char *argv[])
+{
+	struct net_if *iface = net_if_get_default();
+	struct wifi_ps_timeout_params params = { 0 };
+	long timeout_ms = 0;
+	int err = 0;
+
+	context.sh = sh;
+
+	if (argc != 2) {
+		shell_fprintf(sh, SHELL_WARNING, "Invalid number of arguments\n");
+		return -ENOEXEC;
+	}
+
+	timeout_ms = shell_strtol(argv[1], 10, &err);
+
+	if (err) {
+		shell_error(sh, "Unable to parse input (err %d)", err);
+		return err;
+	}
+
+	params.timeout_ms = timeout_ms;
+
+	if (net_mgmt(NET_REQUEST_WIFI_PS_TIMEOUT, iface, &params, sizeof(params))) {
+		shell_fprintf(sh, SHELL_WARNING, "%s failed\n", "setting power save timeout");
```

Comment:
Why is the message formatted such way (i. e. with `%s`), instead of simply passing the full string to the shell? IMO it introduces extra processing overhead with no visible benefit (I don't see where "setting power save timeout" string could be reused somewhere in the system. I'd just use the full string `"Setting power save timeout failed\n"` for the message.